entity-framework/core/what-is-new/ef-core-7.0/whatsnew.md

```diff
@@ -2627,3 +2627,449 @@ info: Microsoft.EntityFrameworkCore.Database.Command[20101]
+SELECT [u].[Id] AS [Author], COALESCE(SUM([d].[DownloadCount]), 0) AS [TotalCost], AVG(CAST([d].[DownloadCount] AS float)) AS [AverageViews], VARP([d].[DownloadCount]) AS [VariancePopulation], VAR([d].[DownloadCount]) AS [VarianceSample], STDEVP([d].[DownloadCount]) AS [StandardDeviationPopulation], STDEV([d].[DownloadCount]) AS [StandardDeviationSample]
+FROM [Downloads] AS [d]
+INNER JOIN [Uploader] AS [u] ON [d].[UploaderId] = [u].[Id]
+GROUP BY [u].[Id]
+```
+
+### Translation of `string.IndexOf`
+
+> [!TIP]
+> The code shown here comes from [MiscellaneousTranslationsSample.cs](https://github.com/dotnet/EntityFramework.Docs/tree/main/samples/core/Miscellaneous/NewInEFCore7/MiscellaneousTranslationsSample.cs).
+
+EF7 now translates <xref:System.String.IndexOf%2A?displayProperty=nameWithType> in LINQ queries. For example:
+
+<!--
+            var query = context.Posts
+                .Select(post => new { post.Title, IndexOfEntity = post.Content.IndexOf("Entity") })
+                .Where(post => post.IndexOfEntity > 0);
+-->
+[!code-csharp[StringIndexOf](../../../../samples/core/Miscellaneous/NewInEFCore7/MiscellaneousTranslationsSample.cs?name=StringIndexOf)]
+
+This query translates to the following SQL when using SQL Server:
+
+```sql
+SELECT [p].[Title], CAST(CHARINDEX(N'Entity', [p].[Content]) AS int) - 1 AS [IndexOfEntity]
+FROM [Posts] AS [p]
+WHERE (CAST(CHARINDEX(N'Entity', [p].[Content]) AS int) - 1) > 0
+```
+
+### Translation of `GetType` for entity types
+
+> [!TIP]
+> The code shown here comes from [MiscellaneousTranslationsSample.cs](https://github.com/dotnet/EntityFramework.Docs/tree/main/samples/core/Miscellaneous/NewInEFCore7/MiscellaneousTranslationsSample.cs).
+
+EF7 now translates <xref:System.Object.GetType?displayProperty=nameWithType> in LINQ queries. For example:
+
+<!--
+            var query = context.Posts.Where(post => post.GetType() == typeof(Post));
+-->
+[!code-csharp[GetType](../../../../samples/core/Miscellaneous/NewInEFCore7/MiscellaneousTranslationsSample.cs?name=GetType)]
+
+This query translates to the following SQL when using SQL Server with TPH inheritance:
+
+```sql
+SELECT [p].[Id], [p].[AuthorId], [p].[BlogId], [p].[Content], [p].[Discriminator], [p].[PublishedOn], [p].[Title], [p].[PromoText]
+FROM [Posts] AS [p]
+WHERE [p].[Discriminator] = N'Post'
+```
+
+Notice that this query returns only `Post` instances that are actually of type `Post`, and not those of any derived types. This is different from a query that uses `is` or `OfType`, which will also return instances of any derived types. For example, consider the query:
+
+<!--
+            var query = context.Posts.OfType<Post>();
+-->
+[!code-csharp[OfType](../../../../samples/core/Miscellaneous/NewInEFCore7/MiscellaneousTranslationsSample.cs?name=OfType)]
+
+Which translates to different SQL:
+
+```sql
+      SELECT [p].[Id], [p].[AuthorId], [p].[BlogId], [p].[Content], [p].[Discriminator], [p].[PublishedOn], [p].[Title], [p].[PromoText]
```

Comment:
Huh, I wonder if it makes sense for us to filter on the discriminator with an `IN ('Post', 'FeaturedPost')` post, rather than fetching everything and filtering on the client...

Reply:
We normally filter by the discriminator, but this is a query for the base type (there are only two types in the model) so it's like the `OfType` isn't there. Perhaps not the best example to show.

Reply:
Ohh I see, makes sense. Yeah, using an intermediate hierarchy type may make this clearer.

Reply:
Yes, we don't filter anything on client.